### PR TITLE
Geomap: Support showing tooltip content on click (not just hover)

### PR DESCRIPTION
**Pull request by ryantxu**
When geomap has a LOT of data, the tooltip (in the current form) can be super overwhelming.  We should fix that for sure, however a simple/common solution is also to only show things when you click.  This PR avoids showing the popup when you are in "tooltip: none" mode.  It also changes the cursor to a pointer so it is more clear that you *can* click!

### CODE REVIEW

**Comment by ryantxu** [comment]
This is structured as "mode" since I think we will also want something like "summary" that can show a collapsed/quick overview for what was matched, but that is more than we need in this PR

File: public/app/plugins/panel/geomap/module.tsx [hunk 1]
```diff
@@ -121,5 +121,17 @@ export const plugin = new PanelPlugin<GeomapPanelOptions>(GeomapPanel)
...
+      })
+      .addRadio({
+        category,
+        path: 'tooltip.mode',
+        name: 'Tooltip',
+        defaultValue: TooltipMode.Details,
+        settings: {
+          options: [
+            { label: 'None', value: TooltipMode.None, description: 'Show contents on click, not hover' },
+            { label: 'Details', value: TooltipMode.Details, description: 'Show popup on hover' },
```
Comment (nmarrs) [inline public/app/plugins/panel/geomap/module.tsx:133]: Might be a little clearer to end user maybe if name was `Tooltip Display` and options were `Hover` vs `Click`? Though that may be a little confusing given user can "click" in hover mode to persist tooltip so 🤷

**Review by nmarrs — APPROVED** [review]
Tested locally and LGTM!

**Comment by drew08t** [comment]
Tested locally, works after adding tooltip option to migration. LGTM!

**Review by drew08t — COMMENTED** [review]
Nice catch, beat me to it!

File: public/app/plugins/panel/geomap/migrations.test.ts [hunk 2]
```diff
@@ -56,6 +56,9 @@ describe('Worldmap Migrations', () => {
             "showZoom": true,
           },
           "layers": Array [],
+          "tooltip": Object {
```
Comment (drew08t) [inline public/app/plugins/panel/geomap/migrations.test.ts:59]: +1

**Review by drew08t — APPROVED** [review]

**Pull request merged**